### PR TITLE
Fixes for prepacking new internal test

### PR DESCRIPTION
**Pull request by hermanventer**
Added more invariants calls to better enforce the global invariant requiring abstract values with non functional builders to have no arguments.

Also added a way to suppress extraneous errors that may arise while traversing additional function bodies to find read/write conflicts. Fixed the initial traversals to report errors and now pass in parameter needed to allow delayed requires inside these functions.

Fixed a bug in generator where it created an abstract value with a non functional builder and some arguments. Then fixed a bug in JSON.parse where it special cased abstract values produced the wrong way.

Added some logic to test-internal to invoke prepack with specified additional functions if the test case has a particular name. This test has to be run manually since it takes a rather long time to complete.

Added logic to populate CompilerDiagnostic objects with a callStack property so that it is easier to find the ultimate source of a prepack error.

### CODE REVIEW

File: scripts/test-internal.js [hunk 1]
```diff
@@ -72,7 +72,13 @@ function runTest(name: string, code: string): boolean {
       serialize: true,
       speculate: !modelCode,
       sourceMaps: !!sourceMap,
-    });
+    };
+    if (name.endsWith("/bundle.js~"))
```
Comment (NTillmann) [inline scripts/test-internal.js]: This looks fishy.

File: src/serializer/visitors.js [hunk 2]
```diff
@@ -103,7 +103,7 @@ function visitName(state, name, modified) {
   let ref = state.tryQuery(
     () => ResolveBinding(state.realm, name, doesNotMatter, state.val.$Environment),
     undefined,
-    true
```
Comment (NTillmann) [inline src/serializer/visitors.js]: Why this change?

File: scripts/test-internal.js (hunk 1 above)
Comment (hermanventer) [inline scripts/test-internal.js]: It uses a naming convention rather than a runtime option. Not something I want to keep, but fixing the runner to deal with options has not yet made it to the top of my to do list.

File: src/serializer/visitors.js (hunk 2 above)
Comment (hermanventer) [inline src/serializer/visitors.js]: This is not an error collecting pass, so it makes no sense to report errors here.
